landoapi/transplants.py

```diff
@@ -495,43 +606,330 @@ def check_landing_blockers(
+    repo_phid = PhabricatorClient.expect(revision, "fields", "repositoryPHID")
+    repo = transplant_state.stack_data.repositories.get(repo_phid)
+    if not repo or not transplant_state.supported_repos:
+        return None
+
+    # Check if this repository needs an approval from relman.
+    local_repo = transplant_state.supported_repos.get(repo["fields"]["shortName"])
+    if not local_repo or local_repo.approval_required is False:
+        return None
+
+    # Check that relman approval was requested and that the
+    # approval was granted.
+    reviewers = get_collated_reviewers(revision)
+    relman_review = reviewers.get(transplant_state.relman_group_phid)
+    if relman_review is None or relman_review["status"] != ReviewerStatus.ACCEPTED:
+        return (
+            "The release-managers group did not accept the stack: "
+            "you need to wait for a group approval from release-managers, "
+            "or request a new review."
+        )
+
+    return None
+
+
+def check_revision_data_classification(
+    revision: dict, diff: dict, transplant_state: TransplantAssessmentState
+) -> Optional[str]:
+    """Check that the `needs-data-classification` tag is not present on a revision."""
+    if revision_has_needs_data_classification_tag(
+        revision, transplant_state.data_policy_review_phid
+    ):
+        return (
+            "Revision makes changes to data collection and "
+            "should have its data classification assessed before landing."
+        )
+
+
+def blocker_single_landing_repo(
+    transplant_state: TransplantAssessmentState,
+) -> Optional[str]:
+    """Assert that a landing request has a single landing repository."""
+    if not transplant_state.landing_assessment:
+        return None
+
+    repo_phid = None
+    for revision, _diff in transplant_state.landing_assessment.to_land:
+        revision_repo_phid = PhabricatorClient.expect(
+            revision, "fields", "repositoryPHID"
+        )
+        if not repo_phid:
+            repo_phid = revision_repo_phid
+            continue
+
+        if revision_repo_phid != repo_phid:
+            return "Landing path contains multiple repositories."
+
+    if not repo_phid:
+        return "Landing path has no repository specified."
+
+    # Set the landing repo field on the `TransplantAssessmentState`.
+    landing_repo = transplant_state.landable_repos.get(repo_phid)
+    transplant_state.landing_assessment.landing_repo = landing_repo
+
+
+STACK_BLOCKER_LINTS = [
+    # This check needs to be first.
+    blocker_stack_landable,
+    blocker_landing_already_requested,
+    user_block_no_auth0_email,
+    blocker_single_landing_repo,
+]
+
+REVISION_BLOCKER_LINTS = [
+    user_block_scm_level,
+    blocker_unsupported_repo,
+    blocker_open_parents,
+    blocker_closed_revisions,
+    blocker_latest_diffs,
+    check_author_planned_changes,
+    check_diff_author_is_known,
+    check_uplift_approval,
+    check_revision_data_classification,
```

Comment:
may be good to keep the `blocker_` prefix/pattern for ease of identifying which methods are intended as blockers, or perhaps change everything to say `check_*`.